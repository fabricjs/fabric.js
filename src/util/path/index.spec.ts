import { getPathSegmentsInfo } from '.';

describe('Path Utils', () => {
  describe('getPathSegmentsInfo', () => {
    it('operates as expected', () => {
      const data = getPathSegmentsInfo([
        ['M', 50, 50],
        ['Q', 50, 50, 75, 75],
        ['Q', 100, 100, 75, 150],
        ['Q', 50, 200, 225, 175],
        ['Q', 400, 150, 450, 325],
        ['L', 500, 500],
      ]);
      expect(data).toMatchSnapshot();
    });
  });
});
