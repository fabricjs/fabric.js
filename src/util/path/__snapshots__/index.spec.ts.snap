// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Path Utils getPathSegmentsInfo operates as expected 1`] = `
[
  {
    "command": "M",
    "length": 0,
    "x": 50,
    "y": 50,
  },
  {
    "angleFinder": [Function],
    "command": "Q",
    "iterator": [Function],
    "length": 35.35533905932738,
    "x": 50,
    "y": 50,
  },
  {
    "angleFinder": [Function],
    "command": "Q",
    "iterator": [Function],
    "length": 80.4908603478418,
    "x": 75,
    "y": 75,
  },
  {
    "angleFinder": [Function],
    "command": "Q",
    "iterator": [Function],
    "length": 174.94631067949948,
    "x": 75,
    "y": 150,
  },
  {
    "angleFinder": [Function],
    "command": "Q",
    "iterator": [Function],
    "length": 301.55412388505005,
    "x": 225,
    "y": 175,
  },
  {
    "command": "L",
    "length": 182.00274723201295,
    "x": 450,
    "y": 325,
  },
  {
    "length": 774.3493812037316,
    "x": 500,
    "y": 500,
  },
]
`;
